### PR TITLE
ocamlPackages.yaml: 3.0.0 -> 3.1.0

### PR DESCRIPTION
**Pull request by superherointj**
* ocamlPackages.yaml: 3.0.0 -> 3.1.0
  - Fixes failing build after `rresult` upgrade (#165097)
* ocamlPackages.yaml-sexp: remove useDune2
* ocamlPackages.ppx_deriving_yaml: 0.1.0 -> 0.1.1
  - Fixes failing build after `rresult` upgrade (#165097)

### CODE REVIEW

**Comment by vbgl** [comment]
This is hard to review as you’ve broken `fmt` there: #165068.

**Comment by superherointj** [comment]
> This is hard to review as you’ve broken `fmt` there: #165068.

Oh. Will fix. Sorry.

**Comment by superherointj** [comment]
> This is hard to review as you’ve broken `fmt` there: #165068.

I have reverted that commit. Is `fmt fine for you now?

**Comment by vbgl** [comment]
It seems that `yaml` requires OCaml ≥ 4.13. Please set `minimalOCamlVersion` accordingly.

**Comment by superherointj** [comment]
@vbgl I've noticed that you add `minimalOCamlVersion` before `src` (which is fetch/unpack phase). And `minimalOCamlVersion` seems to be related to configure phase. So I happen to add `minimalOCamlVersion` after `src` because of this. The same applies to `duneVersion`. Can we agree to follow the usual nix phases? It seemed more logical (to me at least).

**Comment by vbgl** [comment]
> Can we agree […]

I think it is good to disagree on this kind of matter.

File: pkgs/development/ocaml-modules/yaml/default.nix [hunk 1]
```diff
@@ -1,24 +1,24 @@
 { lib, fetchurl, buildDunePackage, ocaml
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/yaml/default.nix:1]:
```suggestion
{ lib, fetchurl, buildDunePackage
```

**Pull request merged**